### ISSUE TITLE
Can a previous bad backup spoil future backups?

### ISSUE THREAD

**Issue by MKs2w7cBxTRP3e2ivnYBddxc** [borgbackup/borg#4744]
I haven't noticed this issue personally yet, but I'm worried whether it's possible that backup_1 where a VM isn't closed (thus creating a bad backup) could affect backup_2 where the VM was properly closed?

**Comment by ThomasWaldmann** [comment]
No. As each borg backup is logically a full backup, in general a past backup can't spoil a future one. [*]

In your case the VM file will be changed, chunked again and chunks not already present in the repo will be added to the repo.

[*] Disclaimer: while true for the general case, there might be special cases with misbehaved and incorrectly working hard and software.

E.g. one could imagine that after computing the MAC (chunk id) of the correct chunk content data that data gets corrupted (e.g. due to a RAM issue). This issue will go unnoticed until the MAC is compared to the data again (e.g. when reading the data from the repo or doing a repo check with --verify-data). As the MAC is correct, the deduplication "thinks" it already has the correct data while in fact it only has the corrupted version of that data. In that case the past bad backup affects the current or future backup due to deduplication.

This problem is not borg specific - you can't reasonably work with hw or sw that corrupts data.
